src/main/java/io/stargate/sgv2/jsonapi/service/embedding/configuration/EmbeddingProviderConfigStore.java

```diff
@@ -9,15 +9,29 @@ record ServiceConfig(
       String serviceProvider,
       String apiKey,
       String baseUrl,
-      Optional<Class<?>> clazz) {
+      Optional<Class<?>> clazz,
```

Comment:
nit: should this be something more descriptive like... `implementationClass` or such?
And/or Javadoc to briefly explain what it means.